### PR TITLE
Add documentation to Signal

### PR DESCRIPTION
**Pull request by oprypin**

### CODE REVIEW

File: src/signal.cr [hunk 1]
```diff
@@ -72,18 +72,43 @@ require "c/unistd"
...
+# end
+# puts "Ctrl+C will be caught from now on"
+# sleep 3
+#
+# Signal::INT.reset
+# puts "Ctrl+C is back to the OS default action"
+# sleep 3
+# ```
+#
+# Note:
```
Comment (Sija) [inline src/signal.cr]: `NOTE:`?

Comment (oprypin) [inline src/signal.cr]: Looks ugly alone on a line.

Comment (Sija) [inline src/signal.cr]: That's subjective and it's going' to be properly processed by the doc formatter which requires all caps.

**Comment by oprypin** [comment]
Uh ping?
@RX14?

**Pull request merged**